website_sale_preset_carts/models/time_frame.py

```diff
@@ -119,14 +119,20 @@ def generate_sale_orders(self):
 
             unit_lines = frame._prepare_order_lines()
 
+            # we first get the customers with the corresponding subscription
             customers = (
                 self.env['res.partner']
-                    .search([('subscription_id', '=', frame.subscription_id.id),
-                             '|', ('cart_suspended_date', '=', False),
-                                  ('cart_suspended_date', '<=', frame.delivery_date)])
+                    .search([('subscription_id', '=', frame.subscription_id.id)])
             )
 
-            for customer in customers:
-                order = self._create_sale_order(customer, unit_lines)
+            # we then filter based on suspended date if suspended
+            subscribers = customers.filtered(lambda cust: cust.suspend_cart == False or (
+                                             cust.cart_suspended_from > frame.delivery_date 
+                                             or cust.cart_suspended_date < frame.delivery_date
+                                             )
+                                            )
```

Comment:
Ici, je préfère quelque chose comme :
```python
subscribers = customers.filtered(
    lambda cust: not cust.suspend_cart or (
        cust.cart_suspended_from > frame.delivery_date
        or cust.cart_suspended_date < frame.delivery_date
    )
)
```
Au dela de l'indentation, c'est surtout le remplacement de `== False` par `not` que je veux mettre en avant.

Reply:
done !